Sources/TransifexNative/Locales.swift

```diff
@@ -46,43 +61,53 @@ public final class LocaleState : NSObject {
     
     /// The currently selected locale that is being displayed in the app
     @objc
-    public var currentLocale : String {
+    public var currentLocale: String {
         get {
             return currentLocaleProvider.currentLocale()
         }
     }
     
     /// A list of all locales supported in the app
     @objc
-    public private(set) var appLocales : [String]
+    public private(set) var appLocales: [String]
+    
+    /// An array containing the app's locales without the source locale.
+    @objc
+    public private(set) var translatedLocales: [String]
     
     /// The provider object that provides the current locale value
     /// whenever it is requested by the currentLocale property.
-    private var currentLocaleProvider : CurrentLocaleProvider
+    private var currentLocaleProvider: TXCurrentLocaleProvider
+    
+    private static let DEFAULT_SOURCE_LOCALE = "en"
     
     /// Constructor.
     ///
     /// - Parameters:
-    ///   - sourceLocale: the locale of the source language, defaults to "en"
-    ///   - appLocales: a list of all locales supported by the application, defaults to ["en"]
+    ///   - sourceLocale: the locale of the source language, defaults to "en" if no source locale is
+    ///   provided
+    ///   - appLocales: a list of all locales supported by the application, defaults to the source locale
+    ///   if the appLocales list is empty
     ///   - currentLocaleProvider: an object conforming to CurrentLocaleProvider protocol,
-    /// defaults to UserDefaultsLocaleProvider
+    /// defaults to TXPreferredLocaleProvider
     @objc
     public init(sourceLocale: String? = nil,
          appLocales: [String] = [],
-         currentLocaleProvider: CurrentLocaleProvider? = nil
+         currentLocaleProvider: TXCurrentLocaleProvider? = nil
     ) {
-        self.sourceLocale = sourceLocale ?? "en"
+        let sourceLocale = sourceLocale ?? LocaleState.DEFAULT_SOURCE_LOCALE
+        self.sourceLocale = sourceLocale
         if appLocales.count == 0 {
-            self.appLocales = [ self.sourceLocale ]
+            self.appLocales = [ sourceLocale ]
         }
         else {
             // Make sure we filter all duplicate values
             // by converting the array to a Set and back
             // to an array.
             self.appLocales = Array(Set(appLocales))
```

Comment:
I think we should also force `appLocales` to include the source locale as well, inserting it at pos 0 if it's doesn't exist at all.

Reply:
I refactored the logic there so it's more streamlined now.